Assets/JS/bandsintown.js

```diff
@@ -35,46 +38,78 @@ $(document).ready(function() {
                 const new_artist = $(`<div class="artist" id=${index}></div>`)
                 const new_artist_name = $(`<h3 class=col_1${artist.id}>${artist.name} </h3>`)
                 // const new_col_2 = artist.genres[0] ? new_col_2 = $(`<td class=col_2${artist.id}>${artist.genres[0]}</td>`) : new_col_2 = $(`<td class=col_2${artist.id}>N/A</td>`)
                 const new_artist_image = $(`<div class='col_3${artist.id} image-holder' ><img src=${artist.images[2].url} width='50'></div>`)
                 new_artist.append(new_artist_name, new_artist_image)
                 $('#band-details').append(new_artist)
                 artists_search_results.push(artist.name)
+                artist_id.push(artist.id)
             });
         })
 
     }
 
-    $('#search-form').submit(function(event) {
+    $('#search-form').submit(function (event) {
         event.preventDefault()
         get_results()
     })
     //#endregion
 
     //#region bandsintown
-    $(document).on('click', '.artist', function() {
+    $(document).on('click', '.artist', function () {
+        let artistId = artist_id[parseInt(this.id)]
+        let divId = this.id
+        // let artist = artists_search_results[parseInt(this.id)]
+        $.get({
+            url: `https://api.spotify.com/v1/artists/${artistId}/top-tracks?country=AU`,
+            headers: { 'Authorization': 'Bearer ' + token }
+        }).then(function (dat) {
+            console.log(dat);
+            const tracks = dat.tracks
+            const new_Div = $(`<div id='spotify' class='display_hits'><p class='text-right'>x</p><h5>Top Hits</h5> </div>`)
+            $(`#${divId}`).append(new_Div)
+            $('.display_hits').show()
+            tracks.forEach(track => {
+                track.album.album_type  // may be used if needed
+                let new_hit = $('<div>')
+                let ex_url = $(`<div><a href=${track.external_urls.spotify} target='_blank'>${track.name}</a></div>`)
+                track.album.release_date // may be used if needed
+                const preview = track.preview_url ? $(`<div><audio controls src=${track.preview_url}></div`) : ''
+                new_hit.append(ex_url, preview)
+
+                $('#spotify').append(new_hit)
+                $('.text-right').click(() => $('.display_hits').remove())
+            })
+
+        })
+
+
+
+
         artists_search_results.forEach((element, index) => {
-            if (parseInt(this.id) == index) {
+            if (parseInt(this.id) === index) {
                 displayBandsInTownData(element)
             }
         });
     })
 
-    function appendArtistInfo(data) {
+    function appendArtistInfo(data, artist) {
         // artist info
         if (data.name) {
             $(".bands-in-town-list").append($("<h2>").text(data.name).attr("style", "margin: 0 0 5px 0"));
-        }
+        } else {
+            $(".bands-in-town-list").append($("<p>").text(artist + " is not in town"))
+        };
```

Comment:
data.name will always be called if the artist exists in bandsintown and doesn't have any events. The else statement should be after data.upcoming_event_count as the error being undefined only occurs when the artist does not have any events
```suggestion
        }
```

Reply:
LGTM